lib/bigtable/cache/FetchJob.js

```diff
@@ -0,0 +1,135 @@
+"use strict";
+
+const Promise = require("bluebird");
+const debug = require("debug")("yildiz:fetchjob");
+
+const redisClient = require("./redisClient");
+
+class FetchJob {
+  
+  constructor(options = {}, yildiz, metrics) {
+
+    this.options = options;
+    this.redisClient = redisClient.create(options, metrics);
+    this.yildiz = yildiz;
+
+    // How long the keys are going to expire in redis
+    this.expireInSec = (this.options.fetchJob || {}).expireInSec || 60 * 60 * 72; //72 hours
+
+    // Fetch Interval for job
+    this.fetchIntervalInSec = (this.options.fetchJob || {}).fetchIntervalInSec || 3;
+
+    // How long the cache is stored until it should be fetched again
+    this.fetchLastAccess = (this.options.fetchJob || {}).fetchLastAccess || 180;
+
+    // Batch size for storing the cache
+    this.fetchBatchSize = (this.options.fetchJob || {}).fetchBatchSize || 10;
+
+    // Limit in retrieving the lastAccess nodeId from redis
+    this.limit = (this.options.fetchJob || {}).limit || 20;
+
+    this._init();
+  }
+
+  _chunk(arr, size){
+
+    const result = [];
+    for (let i = 0; i < arr.length; i += size) {
+        const chunk = arr.slice(i, i+size);
+        result.push(chunk);
+    }
+    return result;
+  }
+
+  async _init(){
+    this.graphAccess = await this.yildiz.getGraphAccess();
+
+    debug("Running job to cache nodes");
+    this._runJob(this.fetchIntervalInSec);
+  }
+
+  async _runJob(intervalInSec) {
+
+    let keys = null;
+    const intervalInMs = intervalInSec * 1000;
+
+    this._tov = setTimeout(() => {
+      this._jobAction(intervalInSec);
```

Comment:
i would prolly try and catch here and rerun later in case of error, also check if there is a proper chain with awaited promises in your try block, so that you actually catch the errors

Reply:
I tried to catch it as promise, if you take a look at the function itself